osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -401,8 +401,21 @@ private void updateRank()
 
             ScoreRank newRank = RankFromScore(Accuracy.Value, ScoreResultCounts);
             ScoreRank newMaxRank = RankFromScore(MaximumAccuracy.Value, ScoreResultCounts);
+
+            //To prevent the minimum rank got into >S, before completed the entire map,
+            //So that in here it fill the rest of the part with miss.
+            // (i.e. simulating the worse situation of all misses)
+            //ref: https://github.com/ppy/osu/pull/28614#discussion_r1670392366
+            if (ScoreResultCounts.TryGetValue(HitResult.Miss, out int currentMissCounts))
+                ScoreResultCounts[HitResult.Miss] = MaxHits - JudgedHits + currentMissCounts;
+            else
+                ScoreResultCounts[HitResult.Miss] = MaxHits - JudgedHits;
```

Comment:
Any reason not to create a temporary dictionary for this? This is too weird to read as-is.

Also remove the link to the discussion thread, there's no need for it. If anything, one can blame the comment and reach back to this PR.

Reply:
I think creating a temporary dictionary is kinda expensive as any update will trigger this update. And this is syncronize, so i do think it won't effect the outputs

Reply:
Then create it once and re-use it.

Reply:
Implemented in [normalid-awa:features/minimum-and-maximum-rank-display#f8a8668](https://github.com/ppy/osu/pull/28614/commits/f8a8668ac0a2323b09a41ccaf83ee5988741ebe3)